src/middlewared/middlewared/plugins/failover.py

```diff
@@ -435,6 +435,22 @@ async def get_disks_local(self):
         except Exception:
             self.logger.error('Unhandled exception in get_disks_local', exc_info=True)
 
+    @private
+    def mismatch_nics(self):
+        """Determine if NICs match between both controllers."""
+        result = False
+        try:
+            local_nics = self.middleware.call_sync('interface.get_nic_names')
```

Comment:
putting `local_nics =` call inside the `try: except` means that your error message "Unable to call interface.get_nic_names on remote" could be wrong.